graphql-server/src/utils/oncotree.ts

```diff
@@ -110,3 +112,50 @@ async function getOncotreeCodesFromNeo4j() {
     await session.close();
   }
 }
+
+/**
+ * Include cancer type fields in the search query indirectly by searching the Oncotree cache
+ * for matching cancerType or cancerTypeDetailed values. If a match is found, add the corresponding
+ * Oncotree code to the original search query.
+ */
+export function includeCancerTypeFieldsInSearch(
+  where: SampleWhere,
+  oncotreeCache: NodeCache
+) {
+  const customWhere = { ...where };
+
+  const sampleMetadataFilters =
+    customWhere?.hasMetadataSampleMetadata_SOME?.OR ||
+    customWhere?.OR?.find((filter) => filter.hasMetadataSampleMetadata_SOME)
+      ?.hasMetadataSampleMetadata_SOME?.OR;
```

Comment:
Extracts the filters. The resulting `sampleMetadataFilters` might look like this, for example:
```js
[
  {
    "cmoSampleName_CONTAINS": "C-8W908N"
  },
  {
    "importDate_CONTAINS": "C-8W908N"
  },
  ...
]
```

Because each view's `where` is structured differently, we handle each case separately in a logical OR chain. Given `const var = x || y || z;`; if `x` is truthy, `var` gets assigned `x`; if `x` is falsy, we check the truthiness of `y`, and so on.

In this code, `customWhere?.hasMetadataSampleMetadata_SOME?.OR` handles the Request Samples view, and the latter handles the general Samples page. We'll need modify this line to handle the Patient Samples view and potentially the Cohort Samples view.